templates/core/about/builds.html

```diff
@@ -3,77 +3,97 @@
+        dependencies for building the library, but not for building the documentation.
+    </p>
+
+    <h4>Detecting Docs.rs from <code>Cargo.toml</code></h4>
+    <p>
+        Doc.rs recognizes <a href="metadata">package metadata</a> in your <code>Cargo.toml</code>.
+        This can be useful to configure the compilation environment, e.g.:
+        <code>
+            [package.metadata.docs.rs]
+            rustc-args = ["--cfg", "docsrs"]
+        </code>
+        Passing this flag sets <code>#[cfg(docsrs)]</code> (not to be confused with <code>#cfg(doc)</code>) for conditional compilation.
+    </p>
+
+    <h4>Testing documentation builds locally</h4>
+    {%- set build_subcommand = docsrs_repo ~ "/blob/master/README.md#build-subcommand" -%}
+    <p>
+        The Docs.rs <a href="{{ build_subcommand | safe }}">README</a> describes how to build
+        unpublished crate documentation locally using the same build environment as the Docs.rs build agent.
+    </p>
+
+    <h3>Diagnosing failed builds</h3>
+
+    <h4>Missing dependencies</h4>
+    <p>
+        Missing dependencies are a common reason for failed builds.
+        Docs.rs dependencies are managed through
+        <a href="https://github.com/rust-lang/crates-build-env">crates-build-env</a>.
+        See <a href="https://forge.rust-lang.org/docs-rs/add-dependencies.html">Forge</a> for how to add a dependency.
+    </p>
+
+    <h4>Write attempt on read-only directories</h4>
+    <p>
+        Most of the sandbox is a read-only file system, including the source directory of your crate and its dependencies.
+        If your <code>build.rs</code> generates files that are relevant for documentation, consider writing to the <a href="https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates">cargo output directory</a>, passed in the environment variable <code>OUT_DIR</code>.
+    </p>
+
+    <h4>Hitting sandbox limits</h4>
+    <p>
+        All the builds on Docs.rs are executed inside a sandbox with limited
+        resources. If a build fails because it hit limits please
+        <a href="{{ docsrs_repo | safe }}/issues/new/choose">open an issue</a>
+        to get them increased for your crate.
+        Since our build agent has finite resources, we have to consider each case indidually.
+        However, network access will not be enabled for any crate.
+        The current limits are:
+    </p>
+
+    {{ macros::crate_limits(limits=limits) }}
+
+    <h4>Something else went wrong</h4>
+    <p>
+        Please <a href="{{ docsrs_repo | safe }}/issues">file an issue</a> if you have trouble building a crate on Docs.rs.
+    </p>
+
+    <h3>Requesting rebuilds</h3>
+    <p>
+        Some build failures can be fixed by Docs.rs.
+        For example, your crate may fail to build because of a missing dependency or a bug in Docs.rs.
+        You can request a rebuild by <a href="{{ docsrs_repo | safe }}/issues">file an issue</a>.
+    </p>
```

Comment:
These three sections are also incredibly similar and separating them is fairly redundant, it could be one paragraph that follows the scaffold of:

- We build in a sandbox, these are the limits
- If you hit those limits, they can be fixed (but no guarantee since our machine can only handle so much)
    - Networking will never be given to a crate
- If you encounter an issue that falls under these criteria, please open one

Reply:
I kept the "other failures" section, since it always feels nice if other problems are allowed ;)